### ISSUE TITLE
Elastalert 'spike' rule alerting on 0 events being greater than 0 events

### ISSUE THREAD

**Issue by m081072** [Yelp/elastalert#455]
I am using elastalert HEAD as of today.

I am using this rule:

    es_host: *******
    es_port: 443
    use_ssl: True
    name: Mike Learning Two
    type: spike
    index: cwl-*
    threshold: 2
    timeframe:
      minutes: 1
    spike_height: 2
    spike_type: "up"
    filter:
    - query:
        query_string:
          query: "status:404"
    alert:
    - "debug"

And it does indeed detect spikes.  But sometimes it alerts with this message:

    INFO:elastalert:Alert for Mike Learning Two at 2016-03-30T08:27:52.137927Z:
    INFO:elastalert:Mike Learning Two

    An abnormal number (0) of events occurred around 2016-03-30 08:27 UTC.
    Preceding that time, there were only 0 events within 0:01:00

    @timestamp: 2016-03-30T08:27:52.137927Z
    reference_count: 0
    spike_count: 0

Am I doing something wrong or is this a bug?

**Comment by Qmando** [comment]
Yeah. I guess this is sort of a bug. You should set either ``threshold_ref`` or ``threshold_cur`` instead of ``threshold``. The example rule is outdated and incorrect and not setting either value should result in an error.

**Comment by tanxiaolong** [comment]
@mike-hogan  @Qmando 
hi~
i dont know why my error said "```
NotFoundError: TransportError(404, u'index_not_found_exception', u'no such index')
INFO:elastalert:Ran tanxiaolongspiketest from 2016-08-22 17:39 CST to 2016-08-22 17:40 CST: 1389 query hits, 0 matches, 0 alerts sent
INFO:elastalert:Sleeping for 59 seconds
```"
i'm sure that a index named "logstash-sso-2016.08.22" in es..
could you give me some tips?

**Comment by tanxiaolong** [comment]
@mike-hogan  do you get alart mail after change `threshold` to `threshold_ref` or `threshold_cur` ?? 
i got nothing..

**Comment by chriseatworld** [comment]
Hey, I have the same trouble. I get Notification with the following configuration:

name: test
type: spike
index: logstash-*
treshold_cur: 5
timeframe:
  hours: 1
spike_height: 4
spike_type: "up"
filter:
- query:
    query_string:
      query: "http_status:[500 TO 599]"
alert:
- "slack"

ProcessController:  INFO:elastalert:Queried rule test from 2018-02-05 13:43 UTC to 2018-02-05 13:43 UTC: 0 / 0 hits
ProcessController:  INFO:elastalert:Alert sent to Slack
ProcessController:  INFO:elastalert:Ran test from 2018-02-05 13:42 UTC to 2018-02-05 13:43 UTC: 0 query hits (0 already seen), 1 matches, 1 alerts sent

**Comment by Qmando** [comment]
This can legitimately happen. You probably got at least 5 hits in the hour before this, and a number of events at exactly 12:42 UTC that suddenly went out of the reference window, causing an alert to trigger.

Please see the documentation on http://elastalert.readthedocs.io/en/latest/ruletypes.html?highlight=spike#spike

If you don't want this happening, maybe add ``threshold_ref`` too.

**Comment by pezzak** [comment]
Faced with the same issue too. 
Steps to reproduce:
- Start elastalert
- got 3 matches
- do nothing
And then alert sending every X times:

```
INFO:elastalert:Ran 4xx_errors from 2018-03-06 18:48 MSK to 2018-03-06 18:49 MSK: 0 query hits (0 already seen), 0 matches, 0 alerts sent
INFO:elastalert:Sleeping for 14.963879 seconds
INFO:elastalert:Queried rule 4xx_errors from 2018-03-06 18:48 MSK to 2018-03-06 18:49 MSK: 0 / 0 hits
INFO:elastalert:Alert sent to Slack
INFO:elastalert:Ran 4xx_errors from 2018-03-06 18:48 MSK to 2018-03-06 18:49 MSK: 0 query hits (0 already seen), 1 matches, 1 alerts sent
INFO:elastalert:Sleeping for 14.320891 seconds
INFO:elastalert:Queried rule 4xx_errors from 2018-03-06 18:48 MSK to 2018-03-06 18:49 MSK: 0 / 0 hits
INFO:elastalert:Queried rule 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:49 MSK: 0 / 0 hits
INFO:elastalert:Ran 4xx_errors from 2018-03-06 18:48 MSK to 2018-03-06 18:49 MSK: 0 query hits (0 already seen), 0 matches, 0 alerts sent
INFO:elastalert:Sleeping for 14.953911 seconds
INFO:elastalert:Queried rule 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:49 MSK: 0 / 0 hits
INFO:elastalert:Ran 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:49 MSK: 0 query hits (0 already seen), 0 matches, 0 alerts sent
INFO:elastalert:Sleeping for 14.962644 seconds
INFO:elastalert:Queried rule 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:50 MSK: 0 / 0 hits
INFO:elastalert:Ran 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:50 MSK: 0 query hits (0 already seen), 0 matches, 0 alerts sent
INFO:elastalert:Sleeping for 14.963329 seconds
INFO:elastalert:Queried rule 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:50 MSK: 0 / 0 hits
INFO:elastalert:Alert sent to Slack
INFO:elastalert:Ran 4xx_errors from 2018-03-06 18:49 MSK to 2018-03-06 18:50 MSK: 0 query hits (0 already seen), 1 matches, 1 alerts sent
```
rule:
```
name: 4xx_errors
type: spike
index: logstash-short-*
timeframe:
  seconds: 20
spike_height: 2
spike_type: "up"
treshold_ref: 1
filter:
- query:
   query_string:
     query: "status: [400 TO 499]"
```

config:
```
run_every:
  seconds: 15
buffer_time:
  minutes: 1
```

**Comment by pezzak** [comment]
In my case it was a typing error: treshold_ref -> t**h**reshold_ref. And if threshold_ref and threshold_cur not set in config, this alert will occur.

**Comment by ashishkaransingh** [comment]
I am seeing this error too
An abnormal number (0) of events occurred around 2021-03-16 15:54 UTC.
Preceding that time, there were only 0 events within 0:02:00

num_hits: 0
num_matches: 1
placeholder: True
reference_count: 0
spike_count: 0
